examples/petsc/src/libceedsetup.c

```diff
@@ -211,4 +212,125 @@ PetscErrorCode CeedLevelTransferSetup(DM dm, Ceed ceed, CeedInt level, CeedInt n
+  // Element coordinates
+  PetscCall(DMGetCoordinatesLocal(dm, &coords));
+  PetscCall(VecGetArrayRead(coords, &coord_array));
+
+  CeedElemRestrictionCreateVector(elem_restr_x, &x_coord, NULL);
+  CeedVectorSetArray(x_coord, CEED_MEM_HOST, CEED_COPY_VALUES, (PetscScalar *)coord_array);
+  PetscCall(VecRestoreArrayRead(coords, &coord_array));
+
+  // Create the persistent vectors that will be needed in setup and apply
+  CeedVectorCreate(ceed, q_data_size * num_elem * num_qpts, &q_data);
+
+  // Create the QFunction that builds the context data
+  CeedQFunctionCreateInterior(ceed, 1, bp_data.setup_geo, bp_data.setup_geo_loc, &qf_setup_geo);
+  CeedQFunctionAddInput(qf_setup_geo, "x", num_comp_x, CEED_EVAL_INTERP);
+  CeedQFunctionAddInput(qf_setup_geo, "dx", num_comp_x * topo_dim, CEED_EVAL_GRAD);
+  CeedQFunctionAddInput(qf_setup_geo, "weight", 1, CEED_EVAL_WEIGHT);
+  CeedQFunctionAddOutput(qf_setup_geo, "qdata", q_data_size, CEED_EVAL_NONE);
+
+  // Create the operator that builds the quadrature data
+  CeedOperatorCreate(ceed, qf_setup_geo, NULL, NULL, &op_setup_geo);
+  CeedOperatorSetField(op_setup_geo, "x", elem_restr_x, basis_x, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setup_geo, "dx", elem_restr_x, basis_x, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setup_geo, "weight", CEED_ELEMRESTRICTION_NONE, basis_x, CEED_VECTOR_NONE);
+  CeedOperatorSetField(op_setup_geo, "qdata", elem_restr_qd_i, CEED_BASIS_NONE, CEED_VECTOR_ACTIVE);
+
+  // Setup q_data
+  CeedOperatorApply(op_setup_geo, x_coord, q_data, CEED_REQUEST_IMMEDIATE);
+
+  // Set up target vector
+  CeedElemRestrictionCreateVector(elem_restr_u, &rhs, NULL);
+  CeedVectorCreate(ceed, num_elem * num_qpts * num_comp_u, &target);
+  // Create the q-function that sets up the RHS and true solution
+  CeedQFunctionCreateInterior(ceed, 1, bp_data.setup_rhs, bp_data.setup_rhs_loc, &qf_setup_rhs);
+  CeedQFunctionAddInput(qf_setup_rhs, "x", num_comp_x, CEED_EVAL_INTERP);
+  CeedQFunctionAddInput(qf_setup_rhs, "qdata", q_data_size, CEED_EVAL_NONE);
+  CeedQFunctionAddOutput(qf_setup_rhs, "true solution", num_comp_u, CEED_EVAL_NONE);
+  CeedQFunctionAddOutput(qf_setup_rhs, "rhs", num_comp_u, CEED_EVAL_INTERP);
+
+  // Create the operator that builds the RHS and true solution
+  CeedOperatorCreate(ceed, qf_setup_rhs, NULL, NULL, &op_setup_rhs);
+  CeedOperatorSetField(op_setup_rhs, "x", elem_restr_x, basis_x, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(op_setup_rhs, "qdata", elem_restr_qd_i, CEED_BASIS_NONE, q_data);
+  CeedOperatorSetField(op_setup_rhs, "true solution", elem_restr_u_i, CEED_BASIS_NONE, target);
+  CeedOperatorSetField(op_setup_rhs, "rhs", elem_restr_u, basis_u, CEED_VECTOR_ACTIVE);
+
+  // Set up the libCEED context
+  CeedQFunctionContext ctx_rhs_setup;
+  CeedQFunctionContextCreate(ceed, &ctx_rhs_setup);
+  CeedScalar rhs_setup_data[2] = {R, l};
+  CeedQFunctionContextSetData(ctx_rhs_setup, CEED_MEM_HOST, CEED_COPY_VALUES, sizeof rhs_setup_data, &rhs_setup_data);
```

Comment:
What's the convention we've been using in this repo? I standardized Ratel to `sizeof(thing)` over `sizeof thing` a while ago since we had a mix.

Reply:
I think I copied directly from another function in this file, but I'm not sure on the general convention.

Reply:
No worries if it isn't standardized here. That can be on the list of "nice to tidy some day".